### PR TITLE
Allow authentication and ssl connection

### PR DESCRIPTION
**Pull request by convoi**
I took the fix from @bradhandy to allow ssl connection and authentication.
refactored it somewhat, added tests for the configuration and updated the doc..

please note that there are multiple ways to specify the ssl values:
-  directly in your application.conf
-  via an included secondary conf (typesafe config allows include)
-  via JVM system properties [http://docs.oracle.com/javase/6/docs/technotes/guides/security/jsse/JSSERefGuide.html#CustomizingStores]

SSL and auth are completely optional.

BTW: thanks for taking over the development with actual releases..

Justin

### CODE REVIEW

**Comment by felixt-cake** [comment]
👍

File: src/main/scala/de/kaufhof/pillar/config/ConnectionConfiguration.scala [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+object ConfigHelper {
+  implicit def toOptionalConfig(config: Config): OptionalConfig = {
+    new OptionalConfig(config)
+  }
+}
+
+abstract sealed class Auth
+
+case class PlaintextAuth(username: String, password: String) extends Auth
```
Comment (markglh) [inline src/main/scala/de/kaufhof/pillar/config/ConnectionConfiguration.scala:57]: Is this necessary? It only seems to be used in the test.

**Comment by markglh** [comment]
👍

**Comment by j-potts** [comment]
Good stuff, LGTM! 👍

**Comment by lichtsprung** [comment]
Yep, LGTM, too.

Thx @ all  :)

I let the ConnectionConfiguration unchanged for the time being.

Cheers,
Robert

**Pull request merged**